src/box/lua/iproto.c

```diff
@@ -118,7 +138,24 @@ push_iproto_key_enum(struct lua_State *L)
 		{"TXN_ISOLATION", IPROTO_TXN_ISOLATION},
 		{"VCLOCK_SYNC", IPROTO_VCLOCK_SYNC},
 	};
-	push_iproto_constant_subnamespace(L, "key", keys, lengthof(keys));
+	push_iproto_constant_subnamespace(L, "key", iproto_keys,
+					  lengthof(iproto_keys));
+	for (size_t i = 0; i < lengthof(iproto_keys); ++i) {
+		size_t len = strlen(iproto_keys[i].name);
+		char *lowercase = strtolowerdup(iproto_keys[i].name);
+		struct mh_strnu32_node_t translation = {
+			.str = lowercase,
+			.len = len,
+			.hash = mh_strn_hash(lowercase, len),
```

Comment:
Nit (up to you): as hash function you can use Lua built-in hash. Like we use it for tuple field names. Then when you lookup by key name provided by Lua, you already have the hash calculated.

Reply:
> Then when you lookup by key name provided by Lua, you already have the hash calculated.

In the scope of this patch, the translation table is only used for encoding. But AFAIC, this could be applicable in scope of CuriousGeorgiy/tarantool@63eb37c. Applied your suggestion.

Reply:
Thanks for trying! Although this is pointless if you won't use `lua_hashstring()`. You simply replaced one hash function with another, but the idea was to not calculate the hash function for Lua strings at all (it is already calculated and stored in them).

Reply:
Answered below.